src/braket/circuits/circuit.py

```diff
@@ -1043,25 +1045,86 @@ def diagram(self, circuit_diagram_class=AsciiCircuitDiagram) -> str:
         """
         return circuit_diagram_class.build_diagram(self)
 
-    def to_ir(self) -> Program:
+    def to_ir(
+        self,
+        ir_type: IRType = IRType.JAQCD,
+        qubit_reference_type: QubitReferenceType = QubitReferenceType.VIRTUAL,
+    ) -> Union[OpenQasmProgram, JaqcdProgram]:
         """
         Converts the circuit into the canonical intermediate representation.
         If the circuit is sent over the wire, this method is called before it is sent.
 
+        Args:
+            ir_type (IRType): The IRType to use for converting the circuit object to its
+                IR representation.
+            qubit_reference_type (QubitReferenceType): The qubit reference type to use for
+                the IR representation.
+
         Returns:
-            (Program): An AWS quantum circuit description program in JSON format.
+            (Union[OpenQasmProgram, JaqcdProgram]): A representation of the circuit in the
+            `ir_type` format.
+
+        Raises:
+            ValueError: If the supplied `ir_type` is not supported.
         """
-        ir_instructions = [instr.to_ir() for instr in self.instructions]
-        ir_results = [result_type.to_ir() for result_type in self.result_types]
+        if ir_type == IRType.JAQCD:
+            return self._to_jaqcd()
+        elif ir_type == IRType.OPENQASM:
+            return self._to_openqasm(qubit_reference_type)
+        else:
+            raise ValueError(f"Supplied ir_type {ir_type} is not supported.")
+
+    def _to_jaqcd(self) -> JaqcdProgram:
+        ir_instructions = [instr.to_ir(ir_type=IRType.JAQCD) for instr in self.instructions]
+        ir_results = [result_type.to_ir(ir_type=IRType.JAQCD) for result_type in self.result_types]
         ir_basis_rotation_instructions = [
-            instr.to_ir() for instr in self.basis_rotation_instructions
+            instr.to_ir(ir_type=IRType.JAQCD) for instr in self.basis_rotation_instructions
         ]
-        return Program.construct(
+        return JaqcdProgram.construct(
             instructions=ir_instructions,
             results=ir_results,
             basis_rotation_instructions=ir_basis_rotation_instructions,
         )
 
+    def _to_openqasm(self, qubit_reference_type: QubitReferenceType) -> OpenQasmProgram:
+        ir_instructions = ["OPENQASM 3.0;"]
```

Comment:
Move 1090 to 1101 to a _create_openqasm_header function() ?

Reply:
Done